karton/system/system.py

```diff
@@ -32,17 +32,24 @@ def __init__(self, config: Optional[Config]) -> None:
         self.last_gc_trigger = 0.0
 
     def gc_collect_resources(self) -> None:
+        # Collects unreferenced resources left in object storage
         karton_bucket = self.backend.default_bucket_name
         resources_to_remove = set(self.backend.list_objects(karton_bucket))
+        # Note: it is important to get list of resources before getting list of tasks!
+        # Task is created before resource upload to lock the reference to the resource.
         tasks = self.backend.get_all_tasks()
         for task in tasks:
+            if task.status == TaskState.FINISHED:
```

Comment:
Gotta go fast. But it doesn't change anything, because we collect tasks before resources, right?

Reply:
True